sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1271,42 +1295,45 @@ def task_worker(self):
 
                         # TODO: Use fine grained time when the CMIS memory map is available
                         self.port_dict[lport]['cmis_expired'] = now + datetime.timedelta(seconds=self.CMIS_DEF_EXPIRED)
-                        self.port_dict[lport]['cmis_state'] = self.CMIS_STATE_AP_CONF
-                    elif state == self.CMIS_STATE_AP_CONF:
+                        self.port_dict[lport]['cmis_state'] = self.CMIS_STATE_DP_INIT
+                    elif state == self.CMIS_STATE_DP_INIT:
                         if not self.test_config_error(api, host_lanes, ['ConfigSuccess']):
                             if (expired is not None) and (expired <= now):
                                 self.log_notice("{}: timeout for 'ConfigSuccess'".format(lport))
                                 self.reset_cmis_init(lport, retries + 1)
                             continue
 
+                        # Ensure the Datapath is NOT Activated unless the host Tx siganl is good.
+                        # NOTE: Some CMIS compliant modules may have 'auto-squelch' feature where
+                        # the module won't take datapaths to Activated state if host tries to enable
+                        # the datapaths while there is no good Tx signal from the host-side.
+                        if not is_host_tx_ready(lport):
+                            self.log_notice("{} waiting for host tx ready...".format(lport))
+                            continue
+
```

Comment:
is_host_tx_ready flag/signal notification to be checked as well when interface/port is admin shutdown.
Tx laser to be turned off. 
Listen to this notification and perform DP_DEACTIVATE followed by DP_DEINIT on the optical module. 

Please add this use-case handling.

Reply:
i have updated the code please check again